internal/openfga/authz.go

```diff
@@ -0,0 +1,157 @@
+package openfga
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/cloudoperators/heureka/internal/util"
+	"github.com/sirupsen/logrus"
+
+	"github.com/openfga/go-sdk/client"
+	"github.com/openfga/language/pkg/go/transformer"
+)
+
+const (
+	fgaStoreName = "heureka-store"
+
+	userFieldName = "user"
+)
+
+type Authz struct {
+	config *util.Config
+	logger *logrus.Logger
+	client *client.OpenFgaClient
+}
+
+func getAuthModelRequestFromFile(filePath string) (*client.ClientWriteAuthorizationModelRequest, error) {
+	modelBytes, err := os.ReadFile(filePath)
+	if err != nil {
+		return nil, err
+	}
+
+	modelJson, err := transformer.TransformDSLToJSON(string(modelBytes))
+	if err != nil {
+		return nil, err
+	}
+
+	// Unmarshal the JSON into the WriteAuthorizationModelRequest struct
+	var modelRequest client.ClientWriteAuthorizationModelRequest
+	err = json.Unmarshal([]byte(modelJson), &modelRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	return &modelRequest, nil
+}
+
+func NewAuthz(l *logrus.Logger, cfg *util.Config) Authorization {
+	fgaClient, err := client.NewSdkClient(&client.ClientConfiguration{
+		ApiUrl: cfg.OpenFGApiUrl,
```

Comment:
We should use authentication here. Let's use API token auth:
https://github.com/openfga/go-sdk?tab=readme-ov-file#api-token